src/file/functions.rs

```diff
@@ -369,73 +333,72 @@ pub fn copy_from_image(file_copy_params: &[FileCopyFromParams], image_file: &Pat
     Ok(())
 }
 
-fn get_partition_num(image_file: &str, partition: &Partition) -> Result<u8> {
-    match partition {
-        Partition::boot => Ok(1),
-        Partition::rootA => Ok(2),
+fn get_partition_info(image_file: &str, partition: &Partition) -> Result<PartitionInfo> {
+    let mut fdisk = Command::new("fdisk");
+    fdisk
+        .arg("-l")
+        .arg("-o")
+        .arg("Device,Start,End")
+        .arg(image_file);
+    let fdisk_out = exec_cmd_with_output!(fdisk);
+
+    let partition_num = match partition {
+        Partition::boot => 1,
+        Partition::rootA => 2,
         p @ (Partition::factory | Partition::cert) => {
-            let mut fdisk = Command::new("fdisk");
-            fdisk.arg("-l").arg(image_file);
-            let fdisk_out = exec_pipe_cmd!(fdisk);
+            let re = Regex::new(r"Disklabel type: (\D{3})")
+                .context("get_partition_info: failed to create regex")?;
```

Comment:
According to spec building the regex should only fail if the regex is faulty or would cause a too big regex. IMHO this can be safely ~unwrap()~ed here.